### PR TITLE
Test support for bin deployed aspnetcore.dll

### PR DESCRIPTION
**Pull request by Tratcher**
Replaces https://github.com/aspnet/Hosting/pull/879 @JunTaoLuo @muratg @pranavkm

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/IISExpressDeployer.cs [hunk 1]
```diff
@@ -22,6 +23,16 @@ public IISExpressDeployer(DeploymentParameters deploymentParameters, ILogger log
         {
         }
 
+        public bool IsWin8orLater
```
Comment (Eilon) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/IISExpressDeployer.cs]: `Or` please 😄

File: src/Microsoft.AspNetCore.Server.IntegrationTesting/project.json [hunk 2]
```diff
@@ -28,11 +28,15 @@
     "Microsoft.AspNetCore.Testing": "1.1.0-*",
     "Microsoft.Extensions.FileProviders.Embedded": "1.1.0-*",
     "Microsoft.Extensions.Logging.Abstractions": "1.1.0-*",
+    "Microsoft.Extensions.PlatformAbstractions": "1.1.0-*",
```
Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting/project.json]: Do you still need this?

Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting/project.json]: Yes, that's used in the xunit helepers

File: src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/IISExpressDeployer.cs [hunk 3]
```diff
@@ -57,6 +68,34 @@ private CancellationToken StartIISExpress(Uri uri, string contentRoot)
...
+                    if (!IsWin8orLater)
+                    {
+                        // The nupkg build of ANCM does not support Win7. https://github.com/aspnet/AspNetCoreModule/issues/40.
+                        ancmPath = @"%ProgramFiles%\IIS Express\aspnetcore.dll";
+                    }
+                    // Bin deployed by Microsoft.AspNetCore.AspNetCoreModule.nupkg
+                    else if (DeploymentParameters.RuntimeFlavor == RuntimeFlavor.CoreClr
+                        && DeploymentParameters.ApplicationType == ApplicationType.Portable)
+                    {
+                        ancmPath = Path.Combine(contentRoot, @"runtimes\win7-x64\native\aspnetcore.dll");
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/IISExpressDeployer.cs]: aspnetcore.dll is different, its bitness needs to match IIS/Express, not the bitness of the OS or app. That said, I need to follow up with sourabh about this if we're ever going to test mixing bitnesses, I think the non-portable apps will deploy the wrong binary.

**Pull request merged**